src/MongoDB/ServerApi.c

```diff
@@ -0,0 +1,391 @@
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "Server API version \"%s\" is not supported in this driver version", ZSTR_VAL(version));
+		return false;
+	}
+
+	if (*server_api) {
+		mongoc_server_api_destroy(*server_api);
+	}
+
+	*server_api = mongoc_server_api_new(server_api_version);
+
+	if (strict_set) {
+		mongoc_server_api_strict(*server_api, strict);
+	}
+
+	if (deprecation_errors_set) {
+		mongoc_server_api_deprecation_errors(*server_api, deprecation_errors);
+	}
+
+	return true;
+} /* }}} */
+
+/* Initialize the object from a HashTable and return whether it was successful.
+ * An exception will be thrown on error. */
+static bool php_phongo_serverapi_init_from_hash(php_phongo_serverapi_t* intern, HashTable* props) /* {{{ */
+{
+	zval* version;
+	zval* strict;
+	zval* deprecation_errors;
+
+	version = zend_hash_str_find(props, ZEND_STRL("version"));
+	if (!version || Z_TYPE_P(version) != IS_STRING) {
+		// Exception
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "%s initialization requires \"version\" field to be string", ZSTR_VAL(php_phongo_serverapi_ce->name));
+		return false;
+	}
+
+	if ((strict = zend_hash_str_find(props, ZEND_STRL("strict"))) && !ZVAL_IS_NULL(strict)) {
+		if (Z_TYPE_P(strict) != IS_TRUE && Z_TYPE_P(strict) != IS_FALSE) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "%s initialization requires \"strict\" field to be bool or null", ZSTR_VAL(php_phongo_serverapi_ce->name));
+			return false;
+		}
+	}
+
+	if ((deprecation_errors = zend_hash_str_find(props, ZEND_STRL("deprecationErrors"))) && !ZVAL_IS_NULL(deprecation_errors)) {
+		if (Z_TYPE_P(deprecation_errors) != IS_TRUE && Z_TYPE_P(deprecation_errors) != IS_FALSE) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "%s initialization requires \"deprecationErrors\" field to be bool or null", ZSTR_VAL(php_phongo_serverapi_ce->name));
+			return false;
+		}
+	}
+
+	return php_phongo_serverapi_create_libmongoc_object(
+		&intern->server_api,
+		Z_STR_P(version),
+		strict && !ZVAL_IS_NULL(strict),
+		strict && zval_is_true(strict),
+		deprecation_errors && !ZVAL_IS_NULL(deprecation_errors),
+		deprecation_errors && zval_is_true(deprecation_errors)
+	);
+} /* }}} */
+
+/* {{{ proto void MongoDB\Driver\ServerApi::__construct(string $version, [?bool $strict], [?bool $deprecationErrors])
+   Constructs a new ServerApi object */
+static PHP_METHOD(ServerApi, __construct)
+{
+	zend_error_handling     error_handling;
+	php_phongo_serverapi_t* intern;
+	zend_string             *version;
+	zend_bool               strict = 0;
+	zend_bool               strict_null = 1;
+	zend_bool               deprecation_errors = 0;
+	zend_bool               deprecation_errors_null = 1;
+
+	intern = Z_SERVERAPI_OBJ_P(getThis());
+
+	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling);
+
+	ZEND_PARSE_PARAMETERS_START(1, 3)
```

Comment:
Does PHP core indent these blocks? I realize that might not play nice with `clang-format`.

Reply:
Unfortunately, clang-format does not allow us to indent these blocks and changes them right back.